### ISSUE TITLE
unbound: Align *-cache-slabs settings with num-threads

### ISSUE THREAD

**Issue by jpawlowski** [opnsense/plugins#1116]
The [Unbound documentation](https://www.nlnetlabs.nl/documentation/unbound/howto-optimise/) says

> Set `*-slabs` to a power of 2 close to the `num-threads` value. Do this for `msg-cache-slabs`, `rrset-cache-slabs`, `infra-cache-slabs` and `key-cache-slabs`.

However, on my machine I can see the settings do not follow that rule:

``````
num-threads: 4
msg-cache-slabs: 4
rrset-cache-slabs: 4
infra-cache-slabs: 4
key-cache-slabs: 4
``````

According to the official documentation, all `*-cache-slabs` settings shall be 8 in that case, not 4.
This shall be corrected in the code.

**Comment by fichtner** [comment]
doesn't "power of 2 close to 4" qualify 4  as well?

**Comment by fichtner** [comment]
Relevant code: https://github.com/opnsense/core/blob/4942127210e8e14c02a22f54d48c5639a73a2d36/src/etc/inc/plugins.inc.d/unbound.inc#L88

**Comment by jpawlowski** [comment]
Yes it does. However, my interpretation of the text is that the number shall be higher than for num-threads and should not match that number.

**Comment by fichtner** [comment]
The only issue I see with the code is that numprocs is used for num-threads, which seems wrong as it wants the number of CPUs in the system. No?

**Comment by jpawlowski** [comment]
Agree. In that case the code might want to look for hw.ncpu instead of kern.smp.cpus?

Reg "power of two": This [doc](https://calomel.org/unbound_dns.html) is a little more clear in it's interpretation and I believe this is the correct one:

> the number of slabs to use for cache and must be a power of 2 times the number of num-threads set above.

´´´´
num-threads: 1 -> 1*2 -> 1
num-threads: 2 -> 2*2 -> 4
num-threads: 4 -> 4*2 -> 8
...
´´´´
